msal_extensions/libsecret.py

```diff
@@ -0,0 +1,64 @@
+"""Implements a Linux specific TokenCache, and provides auxiliary helper types.
+
+This module depends on pygbject. But `pip install pygobject` would typically fail,
+until you install its dependencies first. For example, on a Debian Linux, you need::
+
+    sudo apt install libgirepository1.0-dev libcairo2-dev python3-dev gir1.2-secret-1
+    pip install pygobject
+
+Alternatively, you could skip Cairo & PyCairo, but you still need to do all these
+(derived from https://gitlab.gnome.org/GNOME/pygobject/-/issues/395)::
+
+    sudo apt install libgirepository1.0-dev python3-dev gir1.2-secret-1
+    pip install wheel
+    PYGOBJECT_WITHOUT_PYCAIRO=1 pip install --no-build-isolation pygobject
+"""
+import gi  # https://pygobject.readthedocs.io/en/latest/getting_started.html
+
+gi.require_version("Secret", "1")  # Would require a package gir1.2-secret-1
+# pylint: disable=wrong-import-position
+from gi.repository import Secret  # Would require a package gir1.2-secret-1
+
+class LibSecretAgent(object):
+    """A loader/saver built on top of low-level libsecret"""
+    # Inspired by https://developer.gnome.org/libsecret/unstable/py-examples.html
+    def __init__(  # pylint: disable=too-many-arguments
+            self,
+            schema_name,
+            attributes,  # {"name": "value", ...}
+            label="",  # Helpful when visualizing secrets by other viewers
+            attribute_types=None,  # {name: SchemaAttributeType, ...}
+            collection=None,  # None means default collection
+            ):  # pylint: disable=bad-continuation
+        self._collection = collection
+        self._attributes = attributes
+        self._label = label
+        self._schema = Secret.Schema.new(schema_name, Secret.SchemaFlags.NONE, {
+            k: (attribute_types or {}).get(k, Secret.SchemaAttributeType.STRING)
+            for k in attributes})
+
+    def save(self, data):
+        """Store data. Returns a boolean of whether operation was successful."""
+        return Secret.password_store_sync(
+            self._schema, self._attributes, self._collection, self._label,
+            data, None)
+
+    def load(self):
+        """Load a password in the secret service, return None when found nothing"""
+        return Secret.password_lookup_sync(self._schema, self._attributes, None)
+
+    def clear(self):
+        """Returns a boolean of whether any passwords were removed"""
+        return Secret.password_clear_sync(self._schema, self._attributes, None)
+
+def verify():
+    """Exception will be raised if libsecret is not functioning"""
+    agent = LibSecretAgent("Schema_Name Testing", {"attr1": "foo", "attr2": "bar"})
+    message = "Test Data"
+    agent.save(message)  # This line would fail when running inside an SSH session
+    if agent.load() != message:
+        raise RuntimeError("libsecret did not perform properly")
```

Comment:
Nit: include more details in the exception message, perhaps an aka.ms link

Reply:
Good idea. Speaking about aka.ms link, I don't currently know how to set one up. Have you already created one? Let's follow up offline on this.
By the way, after testing during weekend (and documenting my findings at line 58), I actually realized that this line 60 will NOT be reachable, because another exception would be raised at line 58. It would be better to refactor this part of code so that the aka.ms link would be available from line 58 too.

Reply:
I ended up providing a url in the error message, and that link points to a wiki page, for now. Resolving this conversation.